tests/unit/plugins/modules/test_keycloak_realm_keys_metadata_info.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from contextlib import contextmanager
+from itertools import count
+
+from ansible.module_utils.six import StringIO
+from ansible_collections.community.general.plugins.modules import \
+    keycloak_realm_keys_metadata_info
+from ansible_collections.community.general.tests.unit.compat import unittest
+from ansible_collections.community.general.tests.unit.compat.mock import patch
+from ansible_collections.community.general.tests.unit.plugins.modules.utils import (
+    AnsibleExitJson, ModuleTestCase, set_module_args)
+
+
+@contextmanager
+def patch_keycloak_api(get_realm_keys_metadata_by_id):
+    """Mock context manager for patching the methods in PwPolicyIPAClient that contact the IPA server
+
+    Patches the `login` and `_post_json` methods
+
+    Keyword arguments are passed to the mock object that patches `_post_json`
+
+    No arguments are passed to the mock object that patches `login` because no tests require it
+
+    Example::
+
+        with patch_ipa(return_value={}) as (mock_login, mock_post):
+            ...
+    """
+
+    obj = keycloak_realm_keys_metadata_info.KeycloakAPI
+    with patch.object(
+        obj, "get_realm_keys_metadata_by_id", side_effect=get_realm_keys_metadata_by_id
+    ) as mock_get_realm_keys_metadata_by_id:
```

Comment:
I think the readability of this code would be improved if this `with``were to stay in only one line. Rename the mock object to something shorter - this symbol name is used only in the very next line, having an explicit name does not add more clarity than the line where it is defined, IMHO.

Reply:
Well, I use black to automatically format my code when saving. I had to disable it in this case, as I would not get the line below 80 characters -- which is the default max line length in black. Unless I am missing something, there is no style guideline for these things in this repo, right?

c6bee82ea1235c29bdf27c10ca1447caa149925b contains a version which looks like this:
```python
@contextmanager
def patch_keycloak_api(side_effect):
    # …
    obj = keycloak_realm_keys_metadata_info.KeycloakAPI
    with patch.object(obj, "get_realm_keys_metadata_by_id", side_effect=side_effect) as obj_mocked:
        yield obj_mocked
```

where the long line is 99 characters wide which seems to be accepted style. :)